aws/resource_aws_lb_listener_rule.go

```diff
@@ -645,11 +887,91 @@ func resourceAwsLbListenerRuleRead(d *schema.ResourceData, meta interface{}) err
 	for i, condition := range rule.Conditions {
 		conditionMap := make(map[string]interface{})
 		conditionMap["field"] = aws.StringValue(condition.Field)
+
+		// Deprecated: remove in next major version of provider
 		conditionValues := make([]string, len(condition.Values))
 		for k, value := range condition.Values {
 			conditionValues[k] = aws.StringValue(value)
 		}
 		conditionMap["values"] = conditionValues
+		// End Deprecated
+
+		if condition.HostHeaderConfig != nil {
+			values := make([]string, len(condition.HostHeaderConfig.Values))
+			for k, value := range condition.HostHeaderConfig.Values {
+				values[k] = aws.StringValue(value)
+			}
+			conditionMap["host_header"] = []interface{}{
+				map[string]interface{}{
+					"values": values,
+				},
+			}
+		}
```

Comment:
I feel we could make all the logic inside the if that is setting the value for `conditionMap["host_header"]` a function that could be better unit tested, here and for other blocks too. Thoughts?

Reply:
`host_header`, `http_request_method`, `path_pattern` and `source_ip` are similar enough that a function could be used. `query_string` and `http_header` are unique though.

Reply:
That sounds like a good deal to me then!

Reply:
I've reduced the boilerplate a bit by using some helper methods.